runtime/proto/service/log_replication.proto

```diff
@@ -52,4 +54,32 @@ enum LogReplicationEntryType {
   SNAPSHOT_REPLICATED = 4;
   SNAPSHOT_END = 5;
   SNAPSHOT_TRANSFER_COMPLETE = 6;
+}
+
+enum ReplicationModel {
+  FULL_TABLE = 0;                 // Full table replication (Single Source to Single Sink = 1:1)
+  ROUTING_QUEUES = 1;             // Routing queue replication (used for entry-level replication) (1:1)
+  LOGICAL_GROUPS = 2;             // Table association to logical group (1:1)
+  MULTI_SOURCE_MERGE = 3;         // Same table replication from multiple sources to single sink (n:1)
+}
+
+message ClientInfoKey {
```

Comment:
I am finding these msg names a bit too similar and getting confused  between them while reading the code.

Can we rename them so the name indicates the table/purpose?

Reply:
I found them confusing at times as well, maybe `RegistrationClientId` + `RegistrationClientInfo` for the registryTable and `SourceClientInfoKey` + `SourceSinksInfoVal` for the sourceMetadataTable might be a better fit if the plan is for these protobufs to be table specific.

Reply:
Is there a comment somewhere that explains the E2E flow using this client?